libraries/AP_RangeFinder/AP_RangeFinder_LeddarOne.cpp

```diff
@@ -130,33 +161,35 @@ bool AP_RangeFinder_LeddarOne::CRC16(uint8_t *aBuffer, uint8_t aLength, bool aCh
  */
 LeddarOne_Status AP_RangeFinder_LeddarOne::send_request(void)
 {
-    uint8_t data_buffer[10] = {0};
-    uint8_t i = 0;
+    uint8_t send_buffer[10] = {0};
+    uint8_t index = 0;
 
     uint32_t nbytes = uart->available();
 
     // clear buffer
     while (nbytes-- > 0) {
         uart->read();
-        if (++i > 250) {
+        if (++index > 250) {
             return LEDDARONE_ERR_SERIAL_PORT;
```

Comment:
I don't understand why reading a lot is an error? In your read state you should just empty the input buffer and accept as many samples as there are - and detect the correct framing from there.